Makefile

```diff
@@ -73,7 +73,9 @@ prepare-release: cross
 
 .PHONY: test
 test:
-	go test -race $(PKGS)
+	# Monkey(patching functions for unit tests) sometimes fails to patch a function if inlining is enabled.
+	# Hence disabling inlining with -gcflags="-N -l"
+	go test -gcflags="-N -l" -race $(PKGS)
```

Comment:
Instead of disabling inlining, why don't we fix it in the code instead? Modifying tests like these may make other issues arise in the future.

Reply:
Ack...
but inlining can be done by the go compiler this disables inlining by the compiler
But yes, the present code I see currently works even without these flags. So I'll disable them for now we can probably add them when we need it